### PR TITLE
Added required parameters to FlexibleSpaceBarSettings

### PR DESCRIPTION
**Pull request by Robiquet**
## Description

Made toolbarOpacity, minExtent and maxExtent required parameters by modifing the FlexibleSpaceBarSettings class.

## Related Issues

https://github.com/flutter/flutter/issues/30574

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 1]
```diff
@@ -262,9 +262,9 @@ class FlexibleSpaceBarSettings extends InheritedWidget {
   /// [FlexibleSpaceBar] widget in its tree for the settings to take affect.
   const FlexibleSpaceBarSettings({
     Key key,
-    this.toolbarOpacity,
-    this.minExtent,
-    this.maxExtent,
+    @required this.toolbarOpacity,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/flexible_space_bar.dart]: The constructor's doc should say: The required [toolbarOpacity], [minExtent], and [maxExtent] parameters must not be null.

File: packages/flutter/lib/src/material/flexible_space_bar.dart [hunk 2]
```diff
@@ -262,9 +262,9 @@ class FlexibleSpaceBarSettings extends InheritedWidget {
...
+    @required this.minExtent,
+    @required this.maxExtent,
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/flexible_space_bar.dart]: Should assert that these parameters are not null, that minExtent < maxExtent, that toolbarOpacity >=  0

**Review by HansMuller — CHANGES_REQUESTED** [review]
This is a nice bit of implementation tightening, thanks for the contribution. Just a few small tweaks are needed.

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 2 above)
Comment (Robiquet) [reply packages/flutter/lib/src/material/flexible_space_bar.dart]: Can I assert minExtent <= maxExtent? If I assert minExtent < maxExtent most of the tests fail

Comment (HansMuller) [reply packages/flutter/lib/src/material/flexible_space_bar.dart]: Sorry about that, yes: minExtent <= maxExtent is correct.

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 1 above)
Comment (Robiquet) [reply packages/flutter/lib/src/material/flexible_space_bar.dart]: Added this, not sure if it's in the correct place though

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 2 above)
Comment (Robiquet) [reply packages/flutter/lib/src/material/flexible_space_bar.dart]: Added these, is it also worth asserting minExtent > 0  and maxExtent > 0?

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/src/material/flexible_space_bar.dart]: That's perfect, thanks

File: packages/flutter/lib/src/material/flexible_space_bar.dart (hunk 2 above)
Comment (HansMuller) [reply packages/flutter/lib/src/material/flexible_space_bar.dart]: [sorry, I was out for a week] Yes, asserting that both extents are > 0 before checking them against currentExtent sounds like a good idea.

**Review by HansMuller — COMMENTED** [review]
With the last tiny change, this PR should be ready to land. 

It would be worth updating the PR's description too: Scaffold and AppBar use the static createSettings method. You can delete the part of the description that begins with "However ...".

**Review by HansMuller — APPROVED** [review]

**Comment by HansMuller** [comment]
@Robiquet - I've made the final tiny change. Thanks again for the contribution!

**Comment by Robiquet** [comment]
@HansMuller Thanks, I've updated the PR description

**Pull request merged**